### PR TITLE
Client Credentials Grant

### PR DESCRIPTION
**Pull request by overheadhunter**
This adds support for the [Client Credentials Grant](https://datatracker.ietf.org/doc/html/rfc6749#section-4.4) type, allowing utilization of this lib for _daemons_ aka _service accounts_ that authenticate without user interaction.

Implementation details:

1. In respect to [RFC 6749, Section 2.3.1](https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1), we use the Basic Authentication Scheme to transmit the client credentials. This is the recommended method and other than using body parameters it MUST be supported by all compliant Authorization Servers.
2. [RFC 7617](https://datatracker.ietf.org/doc/html/rfc7617) amends older standards for Basic Auth in regards of specifying a charset. While UTF-8 is a de-facto standard and a recommended fallback, the encoding of credentials may depend on using the correct charset. So this lib allows specifying a credentials charset.
3. [RFC 6749, Section 6](https://datatracker.ietf.org/doc/html/rfc6749#section-6) specifies that authentication is also required during token refresh:
    
    > If the client type is confidential or the client was issued client credentials [...], the client MUST authenticate with the authorization server
    
    As adhering to this specification is likely to requires an API change, this implementation is currently wilfully violating this specification. Should a token refresh using the Client Credentials Grant type ever be required downstream, we need to add it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +52,15 @@ var httpResponse = oauthClient.authorizationCodeGrant(URI.create("https://login.
 
 If your authorization server doesn't allow wildcards, you can also configure a fixed path (and even port) via e.g. `setRedirectPath("/callback")` and `setRedirectPorts(8080)` before calling `authorize(...)`.
 
+### Client Credentials Grant
```
Comment (SailReal) [inline README.md:55]: Should there be a note in this chapter or in the code snippet that a user can optionally set scopes or should the use have a look into the method signature or method documentation? I can live with either

**Review by SailReal — COMMENTED** [review]
The code looks good to me :+1: 

I'll run some tests and approve it if I don't notice anything.

File: README.md (hunk 1 above)
Comment (overheadhunter) [reply README.md:55]: I can certainly add this, same applies for Authorization Code Grant

Comment (overheadhunter) [reply README.md:55]: added in 8028a7f

**Review by SailReal — APPROVED** [review]
My local tests against Keycloak were also successful.

**Pull request merged**